modules/ROOT/pages/dialog-components.adoc

```diff
@@ -539,6 +539,63 @@ A *textarea* is a multiline text field.
 }
 ----
 
+[[tree]]
+=== tree
+
+A *tree* is a composite component that presents an expandable category tree expandable down one level. It can also include a search and filter field.
+
+*onLeafAction:* Interacting with a *tree* component executes the `+onLeafAction+` callback set in the dialog configuration.
+
+[source,js]
+----
+tinymce.init({
+  selector: "textarea",
+  toolbar: "tree",
+    setup: (ed) => {
+      ed.ui.registry.addButton('tree', {
+      text: 'Tree toolbar button',
+      onAction: () => {
+      ed.windowManager.open({
+      title: 'Tree dialog',
+      buttons: [{type: 'cancel', text: 'Cancel',}],
+      body : {
+        type: 'panel',
+        items: [{
+          type: 'tree',
+          onLeafAction: (id) => {
+            console.log('clicked on item with id', id);
+          },
+          items: [{type: 'directory', id: 'dir', title: 'Directory',
+      menu: {
+        icon: 'image-options',
+          type: 'menubutton',
+          fetch: (success) => success([
+            {type: 'menuitem', text: 'Menu item', onAction: () => {store.add('menuitem');}}
+          ])
+      },
+      children: [
+        {type: 'leaf', title: 'Leaf 3', id: '3',},
+        {type: 'leaf', title: 'Leaf 4', id: '4',
+          menu: { icon: 'image-options', type: 'menubutton', fetch: (success) => success([
+            {type: 'menuitem', text: 'Menu item', onAction: () => {store.add('menuitem');}
+            }
+          ])
+        }
+      },
+    ]
+  },
+  {type: 'leaf', title: 'Leaf 5', id: '5',},
+  {type: 'leaf', title: 'Leaf 6',id: '6',},
+  ]
+  }]
+  }
+  })
+  }
+  })
+  }
+});
```

Comment:
```suggestion
tinymce.init({
  selector: "textarea",
  toolbar: "tree",
  setup: (ed) => {
    ed.ui.registry.addButton('tree', {
      text: 'Tree toolbar button',
      onAction: () => {
        ed.windowManager.open({
          title: 'Tree dialog',
          buttons: [{type: 'cancel', text: 'Cancel',}],
          body : {
            type: 'panel',
            items: [{
              type: 'tree',
              onLeafAction: (id) => {
                console.log('clicked on item with id', id);
              },
              items: [
                {
                  type: 'directory',
                  id: 'dir',
                  title: 'Directory',
                  menu: {
                    icon: 'image-options',
                    type: 'menubutton',
                    fetch: (success) => success([
                      {
                        type: 'menuitem',
                        text: 'Menu item',
                        onAction: () => {
                          store.add('menuitem');
                        }
                      }
                    ])
                  },
                  children: [
                    {
                      type: 'leaf',
                      title: 'Leaf 3',
                      id: '3'
                    },
                    {
                      type: 'leaf',
                      title: 'Leaf 4',
                      id: '4',
                      menu: {
                        icon: 'image-options',
                        type: 'menubutton',
                        fetch: (success) => success([
                          {
                            type: 'menuitem',
                            text: 'Menu item',
                            onAction: () => {
                              store.add('menuitem');
                            }
                          }
                        ])
                      }
                    },
                  ]
                },
                {
                  type: 'leaf',
                  title: 'Leaf 5',
                  id: '5'
                },
                {
                  type: 'leaf',
                  title: 'Leaf 6',
                  id: '6'
                },
              ]
            }]
          }
        });
      }
    });
  },
});
```
I also can't find where `store` is defined, have you tested this configuration to make sure that it works? Cos this looks like it'll throw an error when it tries to find `store`.

Reply:
> have you tested this configuration to make sure that it works?

[Yes](https://fiddle.tiny.cloud/U3iaab).

> Cos this looks like it'll throw an error when it tries to find store.

It throws some violations on loading, but I can’t get it throw an error:
<img width="1126" alt="console for fiddle tiny cloud   U3iaab" src="https://user-images.githubusercontent.com/99639563/229407938-16bb3fc3-56ea-4644-9699-7e1f9bba71ff.png">

Reply:
Thanks for the fiddle. When you unfold down to leaf 4, then click on the three dots, then 'Menu item', the error is thrown. 
![image](https://user-images.githubusercontent.com/89239746/229408290-959860b2-8713-4ede-a2e7-e4b1d6c9de60.png)

Reply:
The `store` variable causes an error and should be removed. It can be replaced by a globally declared object, if it is necessary to include this functionality.

Reply:
I agree, that was my mistake I didn't notice that I left the store variable there. `store.add` calls can be replaced here with `console.log` calls for example to demo the usage.

Reply:
> I agree, that was my mistake I didn't notice that I left the store variable there. store.add calls can be replaced here with console.log calls for example to demo the usage.

I’ve edited the demo javascript accordingly.